src/image_builder.py

```diff
@@ -141,7 +140,14 @@ def image_builder(buildspec, image_types=[], device_types=[]):
         if is_nightly_build_context():
             additional_image_tags.append(tag_image_with_date(image_tag))
 
-        if enable_datetime_tag or build_context != "PR":
+        if build_context != "PR":
+            image_tag = tag_image_with_datetime(image_tag)
```

Comment:
nit: Since `image_tag = tag_image_with_datetime(image_tag)` is common in both the conditions, we can probably reduce this condition to just:

```
if build_context == "PR" and is_build_enabled():
      no_datetime = image_tag
      additional_image_tags.append(no_datetime)

image_tag = tag_image_with_datetime(image_tag)
additional_image_tags.append(image_tag)
```

Reply:
This is not a PR blocker. There are multiple ways to implement a logic. Please feel free to merge.